codemod_yaml/items.py

```diff
@@ -0,0 +1,763 @@
+from __future__ import annotations
+
+import ast
+import enum
+import re
+
+from typing import Any, Iterable, Iterator, Optional, overload, SupportsIndex, Union
+
+from tree_sitter import Node
+
+from .base import Item, YamlStream
+from .style import YamlStyle
+
+
+# TODO haven't figured out whether None can be subclassed
+class Null(Item):
+    def __init__(
+        self,
+        original: Optional[Node] = None,
+        stream: Optional[YamlStream] = None,
+        annealed: bool = False,
+    ):
+        super().__init__(original, stream, annealed)
+
+    @classmethod
+    def from_yaml(self, node: Node, stream: YamlStream) -> "Null":
+        return self(original=node, stream=stream, annealed=False)
+
+    def to_string(self) -> str:
+        return "~"
+
+    def __hash__(self) -> int:
+        return hash(None)
+
+    def __eq__(self, other: object) -> bool:
+        return other is None or isinstance(other, Null)
+
+
+class Integer(int, Item):
+    def __new__(
+        cls,
+        value: int,
+        original: Optional[Node] = None,
+        stream: Optional[YamlStream] = None,
+        annealed: bool = False,
+    ) -> Integer:
+        return int.__new__(cls, value)
+
+    def __init__(
+        self,
+        value: int,
+        original: Optional[Node] = None,
+        stream: Optional[YamlStream] = None,
+        annealed: bool = False,
+    ) -> None:
+        super().__init__(original, stream, annealed)
+
+    @classmethod
+    def from_yaml(cls, node: Node, stream: YamlStream) -> "Integer":
+        assert node.text is not None
+        t = node.text.decode("utf-8")
+        if t[0] == "0" and t[:2] not in ("0o", "0x") and t != "0":
+            t = "0o" + t[1:]
```

Comment:
Is this because a number with a leading zero in yaml is an Octal number? 

What happens if it is `08` or `09`?

Reply:
I believe it will be a SyntaxError.  As long as you aren't trying to edit one of those values, it won't matter.  I haven't implemented floats here yet, but their constants are spelled `.inf` and `.nan` and will need some specialcasing too.  (guess what, `-.inf` is valid but `-.nan` is a string :P )